processor/src/test/java/org/mapstruct/ap/testutil/runner/ProcessorTestExtension.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.testutil.runner;
+
+import java.lang.reflect.Method;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContext;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContextProvider;
+import org.junit.platform.commons.support.AnnotationSupport;
+import org.mapstruct.ap.testutil.ProcessorTest;
+
+/**
+ * The provider of the processor tests based on the defined compilers.
+ *
+ * @author Filip Hrisafov
+ */
+public class ProcessorTestExtension implements TestTemplateInvocationContextProvider {
+
+    @Override
+    public boolean supportsTestTemplate(ExtensionContext context) {
+        return AnnotationSupport.isAnnotated( context.getTestMethod(), ProcessorTest.class );
+    }
+
+    @Override
+    public Stream<TestTemplateInvocationContext> provideTestTemplateInvocationContexts(ExtensionContext context) {
+        Optional<Compiler> withSingleCompiler = AnnotationSupport.findAnnotation(
```

Comment:
Do  we still need to support the `@WithSingleCompiler`. Or has it lost it purpose because you can quickly define an `@ProcessersorTest` with the appropriate compiler? Perhaps, if it has no purpose anymore we should remove it in the future.

Reply:
Honestly, I was thinking the exact same thing. `@WithSingleCompiler` is obsolete since you can do `@ProcessorTest(Compiler.JDK)`. I'll remove it

Reply:
top!